### DISCUSSION TITLE
Neighborhood statistics not showing up in grid_stat run

### DISCUSSION THREAD

**Discussion by davis-wesley-12345** [dtcenter/METplus#1822] [Statistical Computation] [answered]
```
//
// Neighborhood methods
// May be set separately in each "obs.field" entry
//
nbrhd = {
   field      = BOTH;
   vld_thresh = 1;
   shape      = SQUARE;
   width      = [11];
   cov_thresh = [ >=.1];
}
```
I have tried several different ways to set this dictionary up in order to get the neighborhood stat files to populate beyond just putting in the column headers.

My data is all 0's and 1's, nothing in between.  So I'm trying to figure out a neighborhood method that works with discrete data, instead of continuous. Getting the shape, width and convolution thresholds correct has been a challenge.

The python reader needed to interact with the file will be the in the .tar file I dropped off.

`/uv/nrl/veriftools/lib/coamps_reader.py`

It takes in the following arguments in a colon-separated format

/path/to/filename.h5, group_name, yyyymmddhh

In the configs I use environment variables to pass the arguments in like so:

```
fcst = {
      file_type = PYTHON_NUMPY;
field = [ { name = "${READ_MODEL} ${FCST_ARGS}";  } ];

}
```

where 

`${READ_MODEL} = /uv/nrl/veriftools/lib/coamps_reader.py
`
and 

`${FCST_ARGS} = /uv/scripts/CLOUD/DATA_DIR/utc18.h5:coamps_sfc_cu-mask-cbh-co_utc18_tau6_predictor_2018010112-2019123112:2019072612
`

**Comment by davis-wesley-12345** [comment]
still having trouble transferring the .tar file with the data

**Comment by CPKalb** [comment]
Hi Wesley:

What does your output logs from MET look like?  Are there any warnings or errors in that output?  Also, what is the command you are using to call Grid-Stat?  If possible, it would be helpful to see the full configuration file and the messages that are output from MET.  Maybe these are already included in the tar file, but if not, I suspect those are small enough that they could be attached here.  

Also, are you running other output types or just the neighborhood statistics?

Christina

**Reply by davis-wesley-12345** [reply]
The output is just the column names, and blank the rest of the way. Similar to how some of the other discussion were describing their errors, except they were using probabilistic data.

My .stat and .cnt files are being produced just fine.

I didn't get any errors or warnings in my output and I ran with -v 10

Sorry I tried to send this all over yesterday but the ftp kept timing out. I'll try again today.

**Reply by davis-wesley-12345** [reply]
[grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_nbrctc.txt](https://github.com/dtcenter/METplus/files/9626668/grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_nbrctc.txt)
[grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_nbrcnt.txt](https://github.com/dtcenter/METplus/files/9626670/grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_nbrcnt.txt)
[log.txt](https://github.com/dtcenter/METplus/files/9626671/log.txt)
[grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_cnt.txt](https://github.com/dtcenter/METplus/files/9626672/grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_cnt.txt)
[grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_nbrcts.txt](https://github.com/dtcenter/METplus/files/9626673/grid_stat_CLOUD_GRID_STAT_GOES_060000L_20190726_120000V_nbrcts.txt)
[log.txt](https://github.com/dtcenter/METplus/files/9626675/log.txt)
[GridStatConfig_CLOUD_GRID_STAT_2019072612.txt](https://github.com/dtcenter/METplus/files/9626695/GridStatConfig_CLOUD_GRID_STAT_2019072612.txt)

**Reply by CPKalb** [reply]
Okay, let me know when the data is uploaded.  I'll need to take a look at some of the data files, as I don't see anything in the config or log that tells me what is going on.

Christina

**Reply by davis-wesley-12345** [reply]
alright, i'll give the ftp another shot right now. last few times I tried it kept timing out.

**Reply by CPKalb** [reply]
I'm not sure how many files are in the .tar file, but if it's timing out, it could help to separate this into 2 files.

**Reply by davis-wesley-12345** [reply]
alright I got the data uploaded, as well as the file reader.

the argument to the file reader goes like this:

/uv/scripts/CLOUD/DATA_DIR/utc18.h5:coamps_sfc_cu-mask-cbh-co_utc18_tau6_predictor_2018010112-2019123112:2019072612

**Reply by CPKalb** [reply]
HI Wesley:

Thanks.  What is your ${OBS_ARGS} set to?  I'll need that for testing.  Also, I don't have any of your mask files.  I'm not sure if I will need these.  I can try without and see what happens

Christina

**Comment by CPKalb** [comment]
Hi Wesley:

For the Neighborhood statistics, thresholds need to be specified in the MET configuration file.  I'm not sure exactly what variable you are working with, but since your data is all 0's and 1's, I'm guessing one of those will be your threshold.  They should be specified after the name variable in the fcst and obs dictionaries.  Something like this:
cat_thresh = [ <=threshold1, >threshold2, >=threshold3 ];

where threshold1, threshold2, and threshold 3 are numbers.  You can use any number of thresholds desired, and <, <=, >, >= or any combination of those.  If you data is 0 and 1, you may want to use a number slightly less than 1 if you are trying to use 1 as the threshold (sometimes data packing in MET can cause a value of 1 to look like: 0.9999999999999999).

When I add these thresholds, I get output in the neighborhood statistics files.  This [link ](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=cat_thresh#obs)shows how the thresholds should look in a configuration file.

Christina

**Reply by davis-wesley-12345** [reply]
I can't get into the HPC right now to pull my config file in order to show you how I filled that dictionary out, but could you post here what you changed to get it working?

**Comment by CPKalb** [comment, marked as answer]
Hi Wesley:

The only changes I made to the configuration file were to add thresholds to the fcst and obs variables.  I don't know exactly what thresholds you want to use, so you may want to change my numbers or add thresholds of your own.  I did not have your observation file (tested only with the forecast), so you will want to replace that part of the configuration below to point to your observation file. 

fcst = {
file_type = PYTHON_NUMPY;
field = [
      {
        name       = "name = "coamps_reader.py DATA_DIR/utc18.h5:coamps_sfc_cu-mask-cbh-co_utc18_tau6_predictor_2018010112-2019123112:2019072612";
        cat_thresh = [ >=0.999 ];
      }
   ];
}

obs = {
file_type = PYTHON_NUMPY;
field = [
      {
        name       =  "coamps_reader.py observation_file";
        cat_thresh = [ >=0.999 ];
      }
   ];
}

Christina

**Reply by davis-wesley-12345** [reply]
AHA. I wasn't putting cat_thresh in my obs and forecast i think, just the neighborhood stats dictionary. ONce I'm able to get back into my work I'll confirm this as the fix but I'm 99% sure that's the case.
